### PR TITLE
RDK-36267: Improve Efficiency of Network RDKServices APIs

### PR DESCRIPTION
**Pull request by ksanth850**
Reason for change: Improves overall performance of clients apps using RDKServices and reduce number of IARM_Bus_Call
Test Procedure: Fetch interface and getipsettings value from netsrvmgr when first call or whenever onInterfaceIPchange event occurs otherwise use cache value for response.
Risks: High
Signed-off-by: Kumar Santhanam Kumar_Santhanam@comcast.com

### CODE REVIEW

File: Network/Network.cpp [hunk 1]
```diff
@@ -1271,14 +1332,31 @@ namespace WPEFramework
...
+                if (!strcasecmp(onInterface.c_str(), "wifi"))
+                {
+                    m_useIpv6WifiCache = false;
+                }
+                else if (!strcasecmp(onInterface.c_str(), "ethernet"))
+                {
+                    m_useIpv6EthCache = false;
+                }
             }
             if (ipv4Addr != "")
```
Comment (karuna2git) [inline Network/Network.cpp:1349]: can you use !string::empty() function?

File: Network/Network.cpp [hunk 2]
```diff
@@ -1271,14 +1332,31 @@ namespace WPEFramework
         void Network::onInterfaceIPAddressChanged(string interface, string ipv6Addr, string ipv4Addr, bool acquired)
         {
             JsonObject params;
-            params["interface"] = m_netUtils.getInterfaceDescription(interface);
+            string onInterface;
+            params["interface"] = onInterface = m_netUtils.getInterfaceDescription(interface);
             if (ipv6Addr != "")
```
Comment (karuna2git) [inline Network/Network.cpp:1337]: can you use string::empty()

**Review by karuna2git — CHANGES_REQUESTED** [review]

File: Network/Network.cpp (hunk 2 above)
Comment (ksanth850) [reply Network/Network.cpp]: Done

File: Network/Network.cpp (hunk 1 above)
Comment (ksanth850) [reply Network/Network.cpp]: Modified it.

File: Network/Network.cpp [hunk 3]
```diff
@@ -1271,14 +1332,31 @@ namespace WPEFramework
...
+                }
+                else if (!strcasecmp(onInterface.c_str(), "ethernet"))
+                {
+                    m_useIpv6EthCache = false;
+                }
             }
             if (ipv4Addr != "")
             {
                 params["ip4Address"] = ipv4Addr;
+                if (!strcasecmp(onInterface.c_str(), "wifi"))
```
[the remark this answers is not in the document]
Comment (ksanth850) [reply Network/Network.cpp]: Modified it

File: Network/Network.cpp [hunk 4]
```diff
@@ -1271,14 +1332,31 @@ namespace WPEFramework
         void Network::onInterfaceIPAddressChanged(string interface, string ipv6Addr, string ipv4Addr, bool acquired)
         {
             JsonObject params;
-            params["interface"] = m_netUtils.getInterfaceDescription(interface);
+            string onInterface;
+            params["interface"] = onInterface = m_netUtils.getInterfaceDescription(interface);
             if (ipv6Addr != "")
             {
                 params["ip6Address"] = ipv6Addr;
+                if (!strcasecmp(onInterface.c_str(), "wifi"))
```
[the remark this answers is not in the document]
Comment (ksanth850) [reply Network/Network.cpp]: Modified it

File: Network/Network.cpp [hunk 5]
```diff
@@ -547,22 +551,24 @@ namespace WPEFramework
...
-                    }
-                    else
+                    if (getIPSettingsInternal(internal, internalResponse, errCode))
                     {
-                        LOGWARN ("Query to get IPaddress by Family Failed..");
-                        response["ip"] = "";
+                        if (NETWORK_IPADDRESS_ACQUIRED == errCode)
+                        {
+                            response["ip"] = internalResponse["ipaddr"];
+                            m_defInterfaceCache = internalResponse["interface"].String();
```
[the remark this answers is not in the document]
Comment (ksanth850) [reply Network/Network.cpp:566]: For second iteration should go with proper interface value.

**Review by karuna2git — APPROVED** [review]

**Pull request merged**